target/linux/qualcommax/image/ipq50xx.mk

```
@@ -1,3 +1,23 @@
+define Device/glinet_gl-b3000
+ $(call Device/FitImage)
+ $(call Device/UbiFit)
+  SOC := ipq5018
+  DEVICE_VENDOR := GL.iNET
+  DEVICE_MODEL := GL-B3000
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  NAND_SIZE := 128m
+  KERNEL_LOADADDR := 0x41080000
+  DEVICE_DTS_CONFIG := config@mp03.5-c1
+  SUPPORTED_DEVICES:=glinet,gl-b3000 b3000
+  UBINIZE_OPTS := -E 5
+  IMAGES := nand-sysupgrade.bin nand-factory.img 
+  IMAGE/nand-sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/nand-factory.img := append-ubi | nand-factory img | append-metadata
```

I've added a PR to your repository with changes that allow you to use existing functions to generate the image: https://github.com/TheRootED24/openwrt/pull/1
```suggestion
  SUPPORTED_DEVICES := b3000
  BOOT_SCRIPT := glinet_gl-b3000
  IMAGES += factory.img 
  IMAGE/factory.img := append-ubi | qsdk-ipq-factory-nand-glinet | pad-to $$(BLOCKSIZE) | append-metadata
```

---

did you test with uboot ? it wouldn't accept the img you asked me to test. It work in oem webui but not uboot... never tried in openwrt

I did something similar for the time being, but i am going to refactor it again shortly. I am just about finished the new method.